oss-internship-2020/libtiff/test/long_tag.cc

```diff
@@ -0,0 +1,135 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <array>
+#include <vector>
+
+#include "check_tag.h"
+#include "tiffio.h"
+
+/* sapi functions:
+ * TIFFWriteScanline
+ * TIFFOpen
+ * TIFFClose
+ * TIFFGetField (from check_tag.c)
+ * TIFFSetField
+ */
+
+struct LongTag {
+  ttag_t tag;
+  short count;
+  unsigned value;
+};
+
+const std::vector<LongTag> long_tags = {
+    {TIFFTAG_SUBFILETYPE, 1,
+     FILETYPE_REDUCEDIMAGE | FILETYPE_PAGE | FILETYPE_MASK}};
+#define SPP 3  // kSamplePerPixel
+static const unsigned kWidth = 1;
+static const unsigned kLength = 1;
+static const unsigned kBps = 8;
+static const unsigned kRowsPerStrip = 1;
+
+TEST(SandboxTest, LongTag) {
```

Comment:
Wrap the contents of this file in an unnamed namespace. Same for the other test files.